### PR TITLE
Added Order Note which allows customer to add order note

### PR DESCRIPTION
**Pull request by codingdefined**
Based on Issue https://github.com/simplcommerce/SimplCommerce/issues/629

## Description

Created OrderNote column in ShoppingCart_Cart table and created a function which saves the Order Note to the database.

### CODE REVIEW

File: src/Modules/SimplCommerce.Module.Core/wwwroot/site.css [hunk 1]
```diff
@@ -362,6 +362,21 @@ li.nav-item-group, li.nav-item-group a {
...
+    border: 0;
+    border-radius: 0;
+    text-transform: uppercase;
+    margin-left: 10px;
+    cursor: pointer;
+}
+
+.cart-list .order-summary .order-note-form {
+    padding-bottom: 20px;
+
```
Comment (thiennn) [inline src/Modules/SimplCommerce.Module.Core/wwwroot/site.css]:
```suggestion
```

File: src/Modules/SimplCommerce.Module.ShoppingCart/Areas/ShoppingCart/Controllers/CartController.cs [hunk 2]
```diff
@@ -107,6 +107,19 @@ public async Task<ActionResult> ApplyCoupon([FromBody] ApplyCouponForm model)
             return Json(validationResult);
         }
 
+        [HttpPost("cart/save-ordernote")]
+        public async Task<IActionResult> SaveOrderNote([FromBody] SaveOrderNote model)
+        {
+            var currentUser = await _workContext.GetCurrentUser();
+            
+            if (model.OrderNote != "")
```
Comment (thiennn) [inline src/Modules/SimplCommerce.Module.ShoppingCart/Areas/ShoppingCart/Controllers/CartController.cs]: As the OrderNote can be empty, I think we should remove the if statement to allow the customer to delete added OrderNote. Btw, for string, string.IsNullOrWhiteSpace is better than != ""

File: src/Modules/SimplCommerce.Module.ShoppingCart/Services/CartService.cs [hunk 3]
```diff
@@ -136,6 +136,16 @@ public async Task<CouponValidationResult> ApplyCoupon(long userId, string coupon
             return couponValidationResult;
         }
 
+        public async Task SaveOrderNote(long userId, string orderNote)
+        {
+            var cart = _cartRepository.Query().FirstOrDefault(x => x.UserId == userId && x.IsActive);
+            if (cart != null)
```
Comment (thiennn) [inline src/Modules/SimplCommerce.Module.ShoppingCart/Services/CartService.cs]: The cart is expected not to be null here. If it's null there must be something wrong here, so I think we better throw an exception if it null

File: src/Modules/SimplCommerce.Module.ShoppingCart/Areas/ShoppingCart/Controllers/CartController.cs (hunk 2 above)
Comment (codingdefined) [reply src/Modules/SimplCommerce.Module.ShoppingCart/Areas/ShoppingCart/Controllers/CartController.cs]: Thanks, I have done the Suggested Change

File: src/Modules/SimplCommerce.Module.ShoppingCart/Services/CartService.cs (hunk 3 above)
Comment (codingdefined) [reply src/Modules/SimplCommerce.Module.ShoppingCart/Services/CartService.cs]: Throwing ApplicationException if the Cart is Empty

**Pull request merged**